services/server/src/server/services/StorageService.ts

```diff
@@ -230,6 +234,52 @@ export class StorageService {
     }
   }
 
+  /**
+   * This function returns a positive number if the first Status
+   * is better than the second one; 0 if they are the same; or a
+   * negative number if the first Status is worse than the second one
+   */
+  getStatusDiff(status1: Status, status2: Status): number {
+    const statusPriority = [null, "partial", "perfect"];
+    const status1Priority = Math.max(
+      0,
+      statusPriority.findIndex((status) => status === status1),
+    );
+    const status2Priority = Math.max(
+      0,
+      statusPriority.findIndex((status) => status === status2),
+    );
+    return status1Priority - status2Priority;
```

Comment:
I find this function overly complicated.
```suggestion
    const scores = {
       "partial": 1,
       "perfect": 2,
    }
    return scores[status1] - scores[status2]
```